src/main.ts

```diff
@@ -105,24 +97,54 @@ export default class LatexSuitePlugin extends Plugin {
 		this.processSettings();
 	}
 
-	async getSnippets() {
-		const snippetVariables = getSnippetVariables(this.settings.snippetVariables);
-
-		if (!this.settings.loadSnippetsFromFile) {
-			let snippets: Snippet[] = [];
-			try {
-				snippets = await parseSnippets(this.settings.snippets, snippetVariables);
-			} catch (e) {
-				new Notice("Failed to load snippets from settings");
-				console.log("Failed to load snippets from settings:", e);
-			}
-			return snippets;
+	async getSettingsSnippetVariables() {
+		try {
+			return await parseSnippetVariables(this.settings.snippetVariables);
+		} catch (e) {
+			new Notice(`Failed to load snippet variables from settings: ${e}`);
+			console.log(`Failed to load snippet variables from settings: ${e}`);
+			return {};
 		}
-		else {
-			const snippets = await getSnippetsWithinFileOrFolder(this.app.vault, this.settings.snippetsFileLocation, snippetVariables);
+	}
 
-			return snippets;
+	async getSettingsSnippets(snippetVariables: SnippetVariables) {
+		try {
+			return await parseSnippets(this.settings.snippets, snippetVariables);
+		} catch (e) {
+			new Notice(`Failed to load snippets from settings: ${e}`);
+			console.log(`Failed to load snippets from settings: ${e}`);
+			return [];
+		}
+	}
+
+	async getSnippets() {
+		// Get files in snippet/variable folders.
+		// If either is set to be loaded from settings the set will just be empty.
+		const files = getFileSets(this);
+		
+		const snippetVariables =
+			this.settings.loadSnippetVariablesFromFile
+				? await getVariablesFromFiles(this, files)
+				: await this.getSettingsSnippetVariables();
+		
+		// This must be done in either case, because it also updates the set of snippet files
+		const unknownFileVariables = await tryGetVariablesFromUnknownFiles(this, files);
+		if (this.settings.loadSnippetVariablesFromFile) { 
+			// But we only use the values if the user wants them
+			Object.assign(snippetVariables, unknownFileVariables);
 		}
+		 
+		new Notice(`Loaded ${Object.keys(snippetVariables).length} variables`)
+		
+
+		const snippets =
+			this.settings.loadSnippetsFromFile
+				? await getSnippetsFromFiles(this, files, snippetVariables)
+				: await this.getSettingsSnippets(snippetVariables);
+		
+		new Notice(`Loaded ${snippets.length} snippets`)
```

Comment:
```suggestion
		const snippets =
			this.settings.loadSnippetsFromFile
				? await getSnippetsFromFiles(this, files, snippetVariables)
				: await this.getSettingsSnippets(snippetVariables);
		
		new Notice(`Loaded ${snippets.length} snippets and ${Object.keys(snippetVariables).length} snippet variables.`);
```

I like these notices – however, currently the UI gets spammed with them when editing snippets or snippet variables from the settings page.

Can we get them to show only when
- the snippet/variable file/folder locations are modified in the settings, or
- a snippet/variable file is added/changed/deleted?